src/runtime/composables/visual-editing.ts

```diff
@@ -132,6 +138,14 @@ const createSanityHelper = (
         queryStore = createQueryStore(visualEditing, client)
       }
     },
+    visualEditing: {
+      isPreviewing: visualEditingState,
+      inFrame: () => {
+        // Return undefined if on server
+        if (import.meta.server) return undefined;
+        return !!(window.self !== window.top || window.opener)
+      },
+    },
```

Comment:
Won't this always be a constant value? And if so, perhaps we could hoist it out of the composable?